### PR TITLE
Wishlist Functionality added❤

### PR DESCRIPTION
**Pull request by t007rushi**
Features to check:
✅Add to Wishlist 
✅Remove From Wishlist 
😒Adding to a wishlist without login -> alert msg pops

recording

<details><summary>Browser metadata</summary>

```
Path:      /login
Browser:   Chrome 99.0.4844.74 on Windows 10
Viewport:  1536 x 775 @1.25x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-9--game-kart-ecom.netlify.app%2Flogin&os=Windows&os_version=10&browser=Chrome&browser_version=99.0&resolution=1536x775&speed=1&start=true&ref=netlify-source)

</details>

[Open Deploy Preview](https://deploy-preview-9--game-kart-ecom.netlify.app/login) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=623b15cbaf927000094b264c&commentID=623b1883cc3cbc2cb841be42&resolution=resolved)

### CODE REVIEW

**Review by mariyasada — APPROVED** [review]
liked that, you have completed the authentication part

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -2,15 +2,17 @@ import React from "react";
...
-  const { user,setUser  } = useAuth();
+  const { user, setUser } = useAuth();
   const { pathname } = useLocation();
+  const { wishlist } = useWishlist();
   const NavLinkStyles = ({ isActive }) => {
     return {
       fontWeight: isActive ? "bold" : "normal",
-    }; 
+    };
   };
```
Comment (horsemaker) [inline src/components/Header/Header.jsx:16]: Maybe you could make a class for the active nav item.

File: src/pages/WishList/Wishlist.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const { wishlist } = useWishlist();
+
+  return (
+    <div className="wishlist-wrap">
+      {wishlist.length === 0 ? (
+        <>
+          <h1>Wishlist is empty add items here</h1>
+          <Link to="/products">Browse products to add</Link>
+        </>
+      ) : (
```
Comment (horsemaker) [inline src/pages/WishList/Wishlist.jsx:17]: Good check here! 👌

File: src/services/wishlist/addToWishlist.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            const { data } = await axios.post(
+              "api/user/wishlist",
+              { product },
+              {
+                headers: {
+                  authorization: user.tokenVal,
+                },
+              }
+            );
+            setWishlist(data.wishlist);
```
Comment (horsemaker) [inline src/services/wishlist/addToWishlist.js]: Since this is a service, if you could make it work without passing setWishlist here it would be better.

**Review by horsemaker — APPROVED** [review]

**Review by neetukumari4858 — APPROVED** [review]
overall looks good code is readable.

File: src/services/wishlist/addToWishlist.js (hunk 3 above)
Comment (t007rushi) [reply src/services/wishlist/addToWishlist.js]: Thanks for suggestions

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (t007rushi) [reply src/components/Header/Header.jsx:16]: Nice Idea to work on✔

**Pull request merged**